### PR TITLE
nix: modernize + shell.nix + documentation improvements

### PR DESCRIPTION
**Pull request by phip1611**
- [x] update niv dependencies
- [x] introduce `shell.nix`
- [x] update README

Hi @blitz,
would you be so kind and review the nix-related changes? To my knowledge, @l3nkz is not that familiar with Nix yet. I'm sorry if I'm mistaken :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
-============
+## Was ist das?
 
 Dieses Diplomarbeits-Latex-Template stellt ein Skelett für eine
 Diplomarbeit mit zugehöriger Make-Datei zur Verfügung.
 
-Checkout Submodules First
-===================
+## Checkout Submodules First
 In order to use the `$ make checkbiw` script please execute: `git submodule update --init --recursive`
```
Comment (blitz) [inline README.md]: If you want to write this document in German, can you also translate this?

File: README.md [hunk 2]
```diff
@@ -1,17 +1,39 @@
...
+solltest du eine beliebige Bilddatei in ein PDF umwandeln können, wenn
+ImageMagick auf deinem System installiert ist.
+
+An diesen Punkt sollte `make` ein `diplom.pdf` produzieren.
+
+Das Template unterstützt sowohl englischen und deutschen Text. Englisch ist
+standardmäßig eingestellt. Für deutschen Text kann der letzte `\selectlanguage`
+Aufruf in `diplom.tex` einfach weggelassen werden.
+
+## How To Build
```
Comment (blitz) [inline README.md:32]: Markdown typically has a newline between headings and content, and content and code blocks. This is also what you did above.

**Review by blitz — APPROVED** [review]
I've added some changes. See the commit messages for the rationale.

File: shell.nix [hunk 3]
```diff
@@ -1,4 +1,6 @@
-{ pkgs ? import <nixpkgs> { } }:
```
Comment (phip1611) [inline shell.nix]: Good catch

File: shell.nix [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
   nativeBuildInputs = with pkgs; [
-    gnumake
-    tex
-
     # convenience for working with nix files
     niv
     nixpkgs-fmt
   ];
+
+  inputsFrom = [
```
Comment (phip1611) [inline shell.nix:17]: I didn't know this, nice

File: nix/release.nix [hunk 5]
```diff
@@ -5,7 +5,9 @@ let
   tex = import ./tex-toolchain.nix { inherit pkgs; };
   inherit (import sources."gitignore.nix" { inherit (pkgs) lib; }) gitignoreSource;
 in
-pkgs.callPackage ./build.nix {
-  inherit gitignoreSource;
-  inherit tex;
+{
+  pdf = pkgs.callPackage ./build.nix {
```
Comment (phip1611) [inline nix/release.nix:9]: Why did you made this an attribute set again? I removed it as I thought it's not relevant.

File: shell.nix (hunk 3 above)
Comment (blitz) [reply shell.nix]: I have no channels configured on my system, so these tend to fail pretty reliably. :)

File: nix/release.nix (hunk 5 above)
Comment (blitz) [reply nix/release.nix:9]: It's convention and you never now whether someone wants more attributes, e.g. HTML output or something.

Comment (phip1611) [reply nix/release.nix:9]: > It's convention 

Can this be found in the Nix manual?

**Comment by phip1611** [comment]
I think this is ready for review/merge.

ping @l3nkz @jan-tudos @Nils-TUD

**Comment by l3nkz** [comment]
Hm, looks good to me. Though I don't know a lot about nix … @Nils-TUD any comments from your side? If not, then I will merge the PR soon.

**Pull request merged**